rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -120,37 +125,80 @@ def replicate_deleted_system_role(self):
         self._replicate(
             ReplicationEventType.DELETE_SYSTEM_ROLE,
             self._create_metadata_from_role(),
-            self._generate_relations_for_role(),
+            self._generate_relations_for_role(list_all_possible_scopes_for_removal=True),
             [],
         )
 
-    def _generate_relations_for_role(self) -> list[common_pb2.Relationship]:
+    def _check_create_admin_platform_relation(self, role, role_scope, list_relations):
+        """Check system role and create admin and platform system role parent-child relationship."""
+        if role.admin_default:
+            try:
+                parent_uuid = platform_v2_role_uuid_for(
+                    DefaultAccessType.ADMIN,
+                    role_scope,
+                    GlobalPolicyIdService.shared(),
+                )
+
+                create_parent_child_relationship = role_child_relationship(parent_uuid, self.role.uuid)
+                list_relations.append(create_parent_child_relationship)
```

Comment:
We could avoid the parameter mutation here by returning a new list and passing it to `relations.extend` in `_generate_relations_for_role`.